rust/plugin-lib/src/dispatch.rs

```diff
@@ -51,16 +52,55 @@ macro_rules! bail_err {
 /// Handles raw RPCs from core, updating state and forwarding calls
 /// to the plugin,
 pub struct Dispatcher<'a, P: 'a + Plugin> {
-    //TODO: when we add multi-view, this should be an Arc+Mutex/Rc+RefCell
-    views: HashMap<ViewId, View<P::Cache>>,
+    plugin_context: PluginContext<<P>::Cache>,
```

Comment:
In general, when I think of a 'context' as we've used it in this project, I tend to think of it as a _temporary collection of resources that are needed for some particular operation_.

So with that in mind, I would probably not have the plugin context owned by the dispatcher, like this, but rather have it created as needed, borrowing contents from the dispatcher:

```rust
/// Provides access to all active views.
struct PluginContext<'a, C: Cache> {
    views: &'a mut HashMap<ViewId, View<C>>,
}
```

Then it's possible that you would run into borrow check issues creating this, in which case you could destructure self (something like this, I haven't tried to compile):

```rust
fn idle(&mut self, _ctx: &RpcCtx, token: usize) {
        let view_id: ViewId = token.into();
        let Dispatcher { mut plugin, mut views, .. } = self;
        let ctx = PluginContext { views: &mut views };
        self.plugin.idle(view_id, &mut ctx);
     }
```

And this highlights another issue here, which is using the `view_id` as the idle token.

The purpose of this is so that the idle handler can know which view to call idle on in the future. This is nice for plugins that only care about a view at a time. With this change, it isn't necessary; the idle token could be anything.

One option, however, would be to keep the existing behaviour while adding a special 'global' idle token, say `usize::max_size()`; in this case we could keep the current behaviour if the idle token is a view id, but if it was this new token we could call a new plugin fn, `idle_global` (or something) which would be passed this new `PluginContext` struct.

This is just an idea, and there are tradeoffs, but it does have the advantage of keeping the current API pretty much entirely the same, which is nice for plugins that prefer it. 🤷‍♂️